### PR TITLE
Bug/quad 94

### PR DESCRIPTION
**Pull request by suprajkolluri**

### CODE REVIEW

File: Quadriga/src/main/java/edu/asu/spring/quadriga/aspects/RestAccessAspect.java [hunk 1]
```diff
@@ -56,17 +56,17 @@ public Object chkProjectAuthorization(ProceedingJoinPoint pjp, NoAuthorizationCh
...
         boolean haveAccess = true;
 
         // retrieve the logged in User name
         Authentication auth = SecurityContextHolder.getContext().getAuthentication();
-        String userName = auth.getName();
-        
-        if(userName == null || "".equals(userName)){
+        String userName = auth.getName().toLowerCase();
+
+        if (userName == null || "".equals(userName)) {
```
Comment (jdamerow) [inline Quadriga/src/main/java/edu/asu/spring/quadriga/aspects/RestAccessAspect.java]: just use `userName.isEmpyt()`

Comment (jdamerow) [inline Quadriga/src/main/java/edu/asu/spring/quadriga/aspects/RestAccessAspect.java]: instead of `"".equals...`

Comment (jdamerow) [inline Quadriga/src/main/java/edu/asu/spring/quadriga/aspects/RestAccessAspect.java]: just realizing, that was already in there. Please change it anyways :)

**Comment by jdamerow** [comment]
ok to test

**Comment by jdamerow** [comment]
hm, let me think about when to merge this. Once merged, projects that were created the uppercase version of a username can't be accessed anymore. I don't want to break the project for people in the class. We might wait with this until the class is over.

File: Quadriga/src/main/java/edu/asu/spring/quadriga/aspects/RestAccessAspect.java (hunk 1 above)
Comment (suprajkolluri) [inline Quadriga/src/main/java/edu/asu/spring/quadriga/aspects/RestAccessAspect.java]: Just made the change :)

**Comment by suprajkolluri** [comment]
Can't we ask them to update the names in the database?
I think it can be done with one simple query.

**Comment by jdamerow** [comment]
That'll probably create confusion and chaos. Let me think about it a little longer what the best way of doing this is.

**Comment by jdamerow** [comment]
@suprajkolluri, please resolve the conflicts and then I'll merge it.

**Comment by suprajkolluri** [comment]
@jdamerow can you please look at this?

**Comment by jdamerow** [comment]
retest this please

File: Quadriga/src/main/java/edu/asu/spring/quadriga/authentication/AuthenticationSuccessHandlerImpl.java [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+    @Override
+    public void onAuthenticationSuccess(HttpServletRequest request, HttpServletResponse response,
+            Authentication authentication) throws IOException, ServletException {
+        Object obj = authentication.getPrincipal();
+        if (obj instanceof QuadrigaUserDetails) {
+            QuadrigaUserDetails quadrigaUserDetails = (QuadrigaUserDetails) obj;
+            String userName = quadrigaUserDetails.getUsername();
+            quadrigaUserDetails.setUsername(userName.toLowerCase());
+        }
+        response.sendRedirect(request.getContextPath() + "/auth/welcome");
```
Comment (jdamerow) [inline Quadriga/src/main/java/edu/asu/spring/quadriga/authentication/AuthenticationSuccessHandlerImpl.java]: hm, sorry, didn't notice that before, but this leads to users always being redirected to the home page, even if they try to go to another page (e.g. session timed out and they reload the page, they don't want to be redirect to the home page after they lgoged in again, but to where they were before). Can we change that?

Comment (suprajkolluri) [inline Quadriga/src/main/java/edu/asu/spring/quadriga/authentication/AuthenticationSuccessHandlerImpl.java]: Can you take a look now? Changed it.

**Comment by jdamerow** [comment]
retest this please

**Pull request merged**